custom-vu/src/main/kotlin/jces1209/vu/page/PageLoader.kt

```diff
@@ -0,0 +1,81 @@
+package jces1209.vu.page
+
+import org.openqa.selenium.By
+import org.openqa.selenium.WebDriver
+import org.openqa.selenium.support.ui.ExpectedConditions.or
+import org.openqa.selenium.support.ui.ExpectedConditions.visibilityOfElementLocated
+import com.atlassian.performance.tools.jiraactions.api.page.wait
+import org.openqa.selenium.support.ui.ExpectedCondition
+import java.time.Duration
+
+class PageLoader private constructor(
+    private val webDriver: WebDriver,
+    private val correctLocators: List<By>,
+    private val errorLocators: List<By>,
+    private val loadTimeout: Duration
+) {
+
+    fun waitForPageToLoad() {
+        webDriver.wait(
+            condition = (correctLocators + errorLocators).anyElementVisible(),
+            timeout = loadTimeout
+        )
+        assertNoErrors()
+    }
+
+    private fun assertNoErrors() {
+        errorLocators.forEach { locator ->
+            webDriver.findElements(locator).firstOrNull()?.let { element ->
+                throw PageLoaderException(element.text)
+            }
+        }
+    }
+
+    private fun List<By>.anyElementVisible(): ExpectedCondition<Boolean> {
+        return or(*this.map { visibilityOfElementLocated(it) }.toTypedArray())
+    }
+
+    class PageLoaderException(elementText: String) : Exception("Error page detected by \"$elementText\"")
+
+    class Builder(
+        private val webDriver: WebDriver,
+        private val correctLocators: List<By>
+    ) {
+        private var loadTimeout: Duration = Duration.ofSeconds(10)
+        private var errorLocators: List<By> = emptyList()
```

Comment:
`JIRA_CLOUD_ERRORS` seem like a good default

Reply:
OTOH we can instead add `fun cloud()`, which would put all defaults for Cloud.

Reply:
Instead of `fun cloud()` that would put all defaults for cloud I created `fun cloudErrors()` which puts all known cloud error pages. WDYT?